sokutei.h

```diff
@@ -0,0 +1,194 @@
+
+
+char sokutei_counters[SOKUTEI_MAX_COUNTER_COUNT * MAX_SIZE_OF_TYPES * SOKUTEI_MAX_MEASURED_ITERATIONS] = {0};
+int  sokutei_number_of_counters = 0;
+int number_of_iterations = SOKUTEI_MAX_MEASURED_ITERATIONS;
+
+
+char sokutei_get_type_of(const int index){
+    return sokutei_counter_definitions[index][31];
+}
+
+
+int sokutei_get_index_of_counter(const char *counter_name){
+    int i;
+    for(i = 0; i < sokutei_number_of_counters; i++){
+        if(sokutei_string_equals(sokutei_counter_definitions[i], counter_name)){
+            return i;
+        }
+    }
+    return SOKUTEI_NOT_FOUND;
+}
+
+
+void sokutei_set_number_of_iterations(const int iterations){
+
+    if(SOKUTEI_MAX_MEASURED_ITERATIONS < iterations){
+        //TODO call error
+        return;
+    }
+
+    number_of_iterations = iterations;
+}
+
+
+int sokutei_add_counter(const char *counter_name, char type){
+
+    if(type != SOKUTEI_INTEGER_TYPE && type != SOKUTEI_FLOAT_TYPE && type != SOKUTEI_INTERVAL_TYPE){
+        return SOKUTEI_NOT_MATCHING_TYPE;
+    }
+
+    if(sokutei_number_of_counters + 1 >= SOKUTEI_MAX_COUNTER_COUNT){
+        return SOKUTEI_COUNTER_LIMIT_REACHED;
+    }
+
+    sokutei_strcpy(sokutei_counter_definitions[sokutei_number_of_counters], counter_name);
+    sokutei_counter_definitions[sokutei_number_of_counters][SOKUTEI_MAX_COUNTER_NAME_LENGTH + 1] = type;
+    sokutei_number_of_counters++;
+}
+
+
+SOKUTEI_INTEGER_COUNTER_TYPE sokutei_integer_get_counter(const char *counter_name){
+    const int index = sokutei_get_index_of_counter(counter_name);
+
+    if(index == SOKUTEI_NOT_FOUND){
+        return SOKUTEI_NOT_FOUND;
+    }
+    if(sokutei_get_type_of(index) != SOKUTEI_INTEGER_TYPE){
+        //TODO call error
+        return SOKUTEI_NOT_MATCHING_TYPE;
+    }
+
+    SOKUTEI_INTEGER_COUNTER_TYPE *counter = sokutei_counters + (index * MAX_SIZE_OF_TYPES);
+
+    return *counter;
+}
+
+
+SOKUTEI_INTEGER_COUNTER_TYPE sokutei_integer_increment_counter(const char *counter_name, const SOKUTEI_INTEGER_COUNTER_TYPE by){
+    const int index = sokutei_get_index_of_counter(counter_name);
+
+    if(index == SOKUTEI_NOT_FOUND){
+        return SOKUTEI_NOT_FOUND;
+    }
+    if(sokutei_get_type_of(index) != SOKUTEI_INTEGER_TYPE){
+        //TODO call error
+        return SOKUTEI_NOT_MATCHING_TYPE;
+    }
+
+    SOKUTEI_INTEGER_COUNTER_TYPE *counter = (void *)sokutei_counters + (index * MAX_SIZE_OF_TYPES);
+
+    return *counter += by;
+}
+
+
+SOKUTEI_FLOAT_COUNTER_TYPE sokutei_float_get_counter(const char *counter_name){
+    const int index = sokutei_get_index_of_counter(counter_name);
+
+    if(index == SOKUTEI_NOT_FOUND){
+        return SOKUTEI_NOT_FOUND;
+    }
+    if(sokutei_get_type_of(index) != SOKUTEI_FLOAT_TYPE){
+        //TODO call error
+        return SOKUTEI_NOT_MATCHING_TYPE;
+    }
+
+    SOKUTEI_FLOAT_COUNTER_TYPE *counter = sokutei_counters + (index * MAX_SIZE_OF_TYPES);
```

Comment:
Here you rightly avoided the `void *` cast, yet this line will give a compiler warning.

You most definitely need a cast, but you must place it correctly.